File: release/inDrive_upload/Chatbot/apps/backend/ai-service/app/safety.py
import re
from .prompts import DISCLAIMER

ADVICE_PATTERNS = [
    r"\bwhat should i do\b",
    r"\bmy situation\b",
    r"\bcan i sue\b",
    r"\blegal advice\b",
    r"\bshould i take action\b",
    r"\bmy case\b",
    r"\bshould i\b",
]


def is_advice_query(query: str) -> bool:
    q = query.lower()
    return any(re.search(pattern, q) for pattern in ADVICE_PATTERNS)


def refusal_message() -> str:
    return (
        "I can provide general legal information, but I cannot advise on your specific case or actions. "
        "Please contact an official advice service or qualified legal professional for personalised support."
    )


def inject_disclaimer(answer: str) -> str:
    if DISCLAIMER.lower() in answer.lower():
        return answer
    return f"{answer} {DISCLAIMER}"

File: apps/backend/node-api/src/services/safety.js
const ADVICE_PATTERNS = [
  /\bshould i sue\b/i,
  /\bcan i win my case\b/i,
  /\bwhat should i do\b/i,
  /\bam i legally allowed to\b/i,
  /\bcan i claim compensation\b/i,
  /\bcan i sue\b/i,
  /\bcan i win\b/i,
  /\bshould i\b/i,
  /\bwhat should i do next\b/i
];

export const ADVICE_DISCLAIMER =
  "I can provide general legal information, but I cannot give personalised legal advice. Please contact a qualified solicitor, Citizens Advice, or an official legal support service.";

export function detectPersonalAdviceRequest(message) {
  const text = String(message || "").trim();
  if (!text) return false;
  return ADVICE_PATTERNS.some((pattern) => pattern.test(text));
}
